File: src/me/Qball/Wild/Utils/OldFormatConverter.java
package me.Qball.Wild.Utils;

import java.util.ConcurrentModificationException;
import java.util.List;

import org.bukkit.Bukkit;

import me.Qball.Wild.Wild;

public class OldFormatConverter {
	public static Wild wild = Wild.getInstance();
	public static void convert()
	{
		if(!wild.getConfig().getBoolean("Converted"))
		{
			if(wild.getConfig().getStringList("Worlds")==null) {
				Bukkit.getLogger().info("Already Converted");
				wild.getConfig().set("Converted", true);
				wild.saveConfig();
				return;
			}
				List<String> worlds = wild.getConfig().getStringList("Worlds");
				for (String w : worlds) {
					String[] world = w.split(":");
					String worldName = world[0];
					int minX = Integer.parseInt(world[1]);
					int maxX = Integer.parseInt(world[2]);
					int minZ = Integer.parseInt(world[3]);
					int maxZ = Integer.parseInt(world[4]);
					WorldInfo info = new WorldInfo();
					info.setWorldName(worldName);
					info.setMinX(worldName, minX);
					info.setMaxX(worldName, maxX);
					info.setMinZ(worldName, minZ);
					info.setMaxZ(worldName, maxZ);
				}
				for (String w : worlds) {
					wild.getConfig().getList("Worlds").remove(w);
				}
				wild.getConfig().set("Converted", true);
				worlds.clear();
				wild.saveConfig();
			
			
		}
		Bukkit.getLogger().info("Already Converted");
	}

}

File: src/main/kotlin/me/qball/wild/util/OldFormatConverter.kt
package me.qball.wild.util

import me.qball.wild.Wild
import java.util.ConcurrentModificationException

object OldFormatConverter {
    private val plugin = Wild.instance

    fun convert() {
        if(!plugin.config.getBoolean("Converted")) {
            try {
                val worlds = plugin.config.getStringList("Worlds")
                worlds.forEach {
                    val world = it.split(":")
                    val worldName = world[0]
                    val minX = world[1]
                    val maxX = world[2]
                    val minZ = world[3]
                    val maxZ = world[4]
                    val info = WorldInfo(plugin)
                    info.setWorldName(worldName)
                    info.setMinX(worldName, Integer.parseInt(minX))
                    info.setMaxX(worldName, Integer.parseInt(maxX))
                    info.setMinZ(worldName, Integer.parseInt(minZ))
                    info.setMaxZ(worldName, Integer.parseInt(maxZ))
                    plugin.config.getStringList("Worlds").remove(it)
                }
                plugin.config.set("Converted", true)
                worlds.clear()
                plugin.saveConfig()
            }catch(e: Exception) {
                if(e is NullPointerException || e is ConcurrentModificationException) {
                    plugin.logger.info("Already converted")
                    plugin.config.set("Converted", true)
                    plugin.saveConfig()
                }
            }
        }
    }
}


